Assets/Shopify/UIToolkit/ThemeControllerBase.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Shopify.UIToolkit {
+    using UnityEngine;
+    using Shopify.Unity;
+
+    public abstract class ThemeControllerBase : MonoBehaviour {
+        /// <summary>
+        /// The Shop Domain to connect to, in the form of "myshop.myshopify.com"
+        /// </summary>
+        public string ShopDomain;
+
+        /// <summary>
+        /// The Storefront Access Token used to authenticate with the connected shop
+        /// </summary>
+        public string AccessToken;
```

Comment:
Should these properties be `readonly`?

Reply:
With the lazy loading code below it might be safer to restrict the developer from changing these because you can into the scenario where you lazy load the client with one domain/token pair the update it but you'll still be using the old cached client.

Reply:
Hmm this is problematic because readonly properties will not be serialized properly by unity and will be empty on save/restore/play, as well as writing to these fields using the editor....

We could also sanity check the cached client to make sure the shop domain and access token match? then we would be able to produce a new client if these are changed at runtime.